### ISSUE TITLE
[BUG] `GetConfigurationSettings(SettingSelector)` does not set the `ContentType` property

### ISSUE THREAD

**Issue by ie-zero** [Azure/azure-sdk-for-net#38524]
### Library name and version

Azure.Data.AppConfiguration 1.2.0

### Describe the bug

The `GetConfigurationSettings(SettingSelector)` does not set the `ContentType` property i.e. it returns null when the `Fields` on the SettingSelector is set.

Example 
```csharp
var selector = new SettingSelector
{
      KeyFilter = "Operations:FuncApps:Usage:Settings:TestKeyVaultReference",
      LabelFilter = string.Empty,

      // SettingFields.ContentType will return null 
      //Fields = SettingFields.Key | SettingFields.Value | SettingFields.Label | SettingFields.ContentType

      // Works fine
      Fields = SettingFields.All
};
```

### Expected behavior

The `ContentType` property is populated when the `Fields = ... | SettingFields.ContentType`

### Actual behavior

The `ContentType` property is `null`

### Reproduction Steps

1. Create a App Configuration with URI `abc.azconfig.io`
1. Create a KeyVault reference with key "Settings:TestKeyVaultReference" via Azure Portal to a KeyVault secret 
1. Grant access to the accessing use on both App configuration and KeyVault secret. 
1.  Execute the following console app

```csharp
using Azure.Data.AppConfiguration;
using Azure.Identity;

namespace TestSecretRetrieval;

public sealed  class Program
{
    public static void Main()
    {
        var client = new ConfigurationClient(new Uri("https://abc.azconfig.io"), new DefaultAzureCredential());

        var selector = new SettingSelector
        {
            KeyFilter = "Settings:TestKeyVaultReference",
            LabelFilter = string.Empty,

            // SettingFields.ContentType will return null 
            //Fields = SettingFields.Key | SettingFields.Value | SettingFields.Label | SettingFields.ContentType

            // Works fine
            Fields = SettingFields.All
        };
        var settings = client.GetConfigurationSettings(selector);

        // The 'Settings:TestKeyVaultReference' is a valid KeyVault reference and the account which is executed
        //  has access to retrieve the secret.

        foreach (var setting in settings)
        {
            // SettingFields.Key, SettingFields.Value and SettingFields.Label appear to work as expected.
            // The SettingFields.ContentType returns null unless SettingFields.All is specified.

            Console.WriteLine($"{SettingFields.ContentType} = '{setting.ContentType ?? "(null)"}'");
        }
    }
}
```

### Environment

Project is running on .NET 6.0

.NET SDK (reflecting any global.json):
 Version:   6.0.410
 Commit:    28c7c894a3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.410\

Host:
  Version:      7.0.7
  Architecture: x64
  Commit:       5b20af47d9

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  6.0.402 [C:\Program Files\dotnet\sdk]
  6.0.410 [C:\Program Files\dotnet\sdk]
  7.0.100 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by pallavit** [comment]
Thank you for your feedback. Tagging and routing to the team member best able to assist.
/cc: @ShivangiReja , @jsquire

**Comment by kinelski** [comment]
@ie-zero Thank you for reporting this bug. We were able to identify the cause and it has been fixed 
(https://github.com/Azure/azure-sdk-for-net/pull/38917). The fix will be available in our next beta release (1.3.0-beta.3) which is expected to be out in October.

Closing this since the issue has been addressed.

**Comment by ie-zero** [comment]
Thank you for addressing this issue.
